WordPress/Classes/Services/SuggestionService.m

```diff
@@ -91,23 +82,84 @@ - (void)updateSuggestionsForSiteID:(NSNumber *)siteID
     }];
 }
 
+- (Autocompleter *)retrieveAutocompleterForSiteID:(NSNumber *)siteID
+{
+    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
+
+    NSFetchRequest *request = Autocompleter.fetchRequest;
+    [request setPredicate:[NSPredicate predicateWithFormat:@"siteID == %@", siteID]];
+    NSError *error;
+    NSArray *sites = [context executeFetchRequest:request error:&error];
+    if (error) {
+        DDLogError(@"Could not fetch Autocompleter for site %@, error", siteID, error);
+        return nil;
+    } else if (sites.count > 1) {
+        DDLogError(@"Retrieved more than one Autocompleter for site %@", siteID);
```

Comment:
In this case, can't we just return the firstObject?

Reply:
This is a tough call, I'd normally say, wouldn't it be better to fail right here as soon as we know the app is in an unexpected state? In any case, I would leave the logging there.

Reply:
Marking this as resolved because the latest commits get rid of this code.